### PR TITLE
Adds typescript support via babel and updated configs

### PR DESCRIPTION
**Pull request by wejendorp**
**What**:
- Adds TypeScript compilation to the babel build command.
- Adds TypeScript linting to the lint command via eslint-config-tradeshift update.

**Why**:
Align our tools to get rid of tslint and streamline the build of js/ts mixed projects.

**How**:
We've taken the babel config from create-react-app for overrides to typescript.

**Checklist**:

- [ ] Documentation
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: src/config/jest.config.js [hunk 1]
```diff
@@ -15,10 +15,10 @@ const ignores = ['/node_modules/', '/fixtures/', '/__tests__/helpers/', '__mocks
 const jestConfig = {
 	roots: [fromRoot('src')],
 	testEnvironment: ifAnyDep(['webpack', 'rollup', 'react'], 'jsdom', 'node'),
-	collectCoverageFrom: ['src/**/*.js'],
-	testMatch: ['**/__tests__/**/*.js', '**/*.spec.js'],
+	collectCoverageFrom: ['src/**/*.[jt]s?(x)'],
```
Comment (upphiminn) [inline src/config/jest.config.js:18]: should it be `[jt]sx?`, this doesn't match _js, ts_ and matches _tx, jx_

Comment (upphiminn) [reply src/config/jest.config.js:18]: same below

**Review by upphiminn — APPROVED** [review]

File: src/config/jest.config.js (hunk 1 above)
Comment (wejendorp) [reply src/config/jest.config.js:18]: It's the weird glob/extglob syntax. Seems to work afaict, but I'll double check.

https://github.com/micromatch/micromatch#extglobs

**Pull request merged**